src/network/index.ts

```diff
@@ -50,10 +44,21 @@ interface INetworkMetadata {
     }
 }
 
-// TODO: this should probably be cached to avoid pulling for every request
 export async function getMetadata(network: NETWORK_TYPE): Promise<INetworkMetadata> {
-    return fetch(`https://raw.githubusercontent.com/ovrclk/net/master/${network}/meta.json`)
-        .then(res => res.json());
+    if (networkMetadata) {
```

Comment:
this way to cache works. Although it's really a boilerplate code that ideally would needed to be improved. I'd suggest to add some caching lib for this. The easiest one that would already be nice is `lodash` with it's `memoize` fn. Feel free to leave it like this if you like so but if you do, please add a todo based on this comment.